app/src/components/report/DataGrid.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useState } from 'react';
+import Head from 'next/head';
+import PageHeader from '@/components/PageHeader';
+import Container from 'react-bootstrap/Container';
+import EditIcon from '@mui/icons-material/Edit';
+import { Collapse, IconButton } from '@mui/material';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import ExpandLessIcon from '@mui/icons-material/ExpandLess';
+import Box from '@mui/material/Box';
+import {
+    DataGrid,
+    GridRowsProp,
+    GridColDef,
+    GridActionsCellItem,
+    GridRowParams,
+    GridRenderCellParams,
+} from '@mui/x-data-grid';
+import Edit from '@mui/icons-material/Edit';
+
+const expandedRowStyle = {
+    whiteSpace: 'pre-wrap',
+    overflowWrap: 'break-word',
+    paddingBottom: '10px',
+};
+const rows: GridRowsProp = [
+    {
+        id: 1,
+        reporter: 'Min',
+        type: 'Illegal dump',
+        location: 'Seattle',
+        dateReported: '07/01/2023',
+        lastUpdated: '07/01/2023',
+        status: 'Reported',
+    },
+    {
+        id: 2,
+        reporter: 'Andrey',
+        email: 'andrey@gmail.com',
+        phone: '206-XXX-XXXX',
+        description: 'Pls send help...the streets are flooded',
+        type: 'Clogged drain',
+        location: 'Seattle',
+        dateReported: '06/29/2023',
+        lastUpdated: '07/30/2023',
+        status: 'Pending',
+    },
+];
+
+export default function Table() {
+    const [clickedIndex, setClickedIndex] = useState(-1);
+
+    const columns: GridColDef[] = [
+        {
+            field: 'id',
+            headerName: '',
+            width: 80,
+            renderCell: (cellValues: GridRenderCellParams<any>) => {
+                return (
+                    <IconButton
```

Comment:
Could you check if data grid has a built-in row selection feature? It seems unusual that we need to manage this ourselves :)

If we do need to implement it ourselves, could you please add accessibility considerations so screen reader users can use it too? Here's a place to start: https://www.smashingmagazine.com/2017/09/building-inclusive-toggle-buttons/

Reply:
@andreybutenko MUI does have row expansion but it's locked behind the paywall https://mui.com/x/react-data-grid/master-detail/  :,)

I updated the file and add some accessibility features